### PR TITLE
Test-Branch

### PR DESCRIPTION
**Pull request by Naqhid**
This is used to test the Tic Tac Toe Ruby file which we created

### CODE REVIEW

File: spec/board_spec.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+require './lib/board.rb'
+
+RSpec.describe 'testing the board' do
+  it 'should show us an empty board' do
+    expect(Board.new.show_board).to eq("+------------------------+\n| 1 - ❔  2 - ❔  3 - ❔ | \n| 4 - ❔  5 - ❔  6 - ❔ | \n| 7 - ❔  8 - ❔  9 - ❔ | \n+------------------------+")
+  end
+
+  it 'should show us an empty board' do
+    expect(Board.new.show_board).not_to eq("+------------------------+\n| 1 - p  2 - ❔  3 - ❔ | \n| 4 - ❔  5 - ❔  6 - ❔ | \n| 7 - ❔  8 - ❔  9 - ❔ | \n+------------------------+")
+  end
```
Comment (tirthajyoti-ghosh) [inline spec/board_spec.rb]: These tests should be wrapped in a `describe` block. Since both of them are testing the `show_board` method, they should have their own `describe` block.

- [ ] Please wrap them in a `describe` block.

File: spec/board_spec.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+require './lib/board.rb'
+
+RSpec.describe 'testing the board' do
+  it 'should show us an empty board' do
+    expect(Board.new.show_board).to eq("+------------------------+\n| 1 - ❔  2 - ❔  3 - ❔ | \n| 4 - ❔  5 - ❔  6 - ❔ | \n| 7 - ❔  8 - ❔  9 - ❔ | \n+------------------------+")
```
Comment (tirthajyoti-ghosh) [inline spec/board_spec.rb]: Since you are using the `Board.new` in all of your tests in this file, you can use `let` so as to not repeat yourself.

```ruby
let(:board) { Board.new }
```

Then reuse the above code inside all `expect`.

- [ ] Please use `let` to DRY up your code.

File: spec/board_spec.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  end
+
+  it 'should show us an empty board' do
```
Comment (tirthajyoti-ghosh) [inline spec/board_spec.rb]: This statement is the same as the previous test statement. Since this is a negative scenario test, the statement should reflect the actual test you are performing.

```ruby
it 'should not show a filled board' do
  # tests...
end
```

- [ ] Please update the test statements.

File: spec/board_spec.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  end
+
+  context 'update board methods' do
+    it 'should return a board with an input in 5th position' do
+      expect(Board.new.update_board(5, 'x')).not_to eq("+------------------------+\n| 1 - ❔  2 - x  3 - ❔ | \n| 4 - ❔  5 - ❌  6 - ❔ | \n| 7 - ❔  8 - ❔  9 - ❔ | \n+------------------------+")
+    end
+
+    it 'should return a board with an input in 5th position' do
+      expect(Board.new.update_board(5, 'x')).to eq("+------------------------+\n| 1 - ❔  2 - ❔  3 - ❔ | \n| 4 - ❔  5 - ❌  6 - ❔ | \n| 7 - ❔  8 - ❔  9 - ❔ | \n+------------------------+")
+    end
```
Comment (tirthajyoti-ghosh) [inline spec/board_spec.rb]: Both the test statements are the same but both the tests are not. Each test statement should reflect what the actual test is. 

**IMPORTANT!** I have only pointed this out here and above. I have noticed this is the case for most of your tests. So I won't repeat myself. 

- [ ] Please go through all of your tests in the two files and update the test statements so that they reflect the actual test. Eg., the positive scenario test statement should not be the same as the negative scenario test statement.

File: spec/game_spec.rb [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+  it 'it chooses sign for players' do
+    arr = Game.choose_signs
+    expect(arr.sort).to eq(%w[x o].sort)
+  end
+
+  it 'it chooses sign for players' do
+    arr = Game.choose_signs
+    expect(arr.sort).not_to eq(%w[x k].sort)
+  end
```
Comment (tirthajyoti-ghosh) [inline spec/game_spec.rb]: These tests should be wrapped in a `describe` block. Since both of them are testing the `self.choose_signs` method, they should have their own `describe` block.

- [ ] Please wrap them in a `describe` block.


**IMPORTANT!** I have only pointed this out here and above. I have noticed this is the case for some of your tests below. So I won't repeat myself. 

- [ ] Please go through all of the below tests and wrap the tests in a `describe` block that belongs together. Eg., if you have 4 tests for one method, they should be put inside a `describe` block.

File: spec/game_spec.rb [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+    expect(Game.validate_name('Grace')).not_to be(false)
+  end
+
+  it 'validates player name if not less than 3' do
+    expect(Game.validate_name('ab')).to be(false)
+  end
+
+  it 'validates player name if not less than 3' do
+    expect(Game.validate_name('ab')).not_to be(true)
+  end
```
Comment (tirthajyoti-ghosh) [inline spec/game_spec.rb]: I am talking about these tests. They are testing the `validate_name` method. So they should be grouped together.

File: spec/game_spec.rb [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+    expect(player1.sign).to eq('x')
+  end
+
+  it 'get the current player' do
+    expect(player1.name).not_to eq('paul')
+  end
+
+  it 'get the current player' do
+    expect(player1.sign).not_to eq('o')
+  end
```
Comment (tirthajyoti-ghosh) [inline spec/game_spec.rb]: `.name` tests should have their own group and `.sign` tests should have their own group.

**Review by tirthajyoti-ghosh — CHANGES_REQUESTED** [review]
# Changes required 🛑 

## What a remarkable project you have built 💯! This is impressive 🤩👌🧨!

- Your tests are absolutely amazing. You have tested all of the edge cases. Well done!
- The README is very well-structured. 

### But in order to get it approved, we need to work on some changes. Let's see...

- The code linters are informing some errors. Please fix them.

- The PR summary can be improved. Please edit it so that it reflects what changes you have made in this branch - what tests you have added, etc.

- In your README, there should be a section to describe how to run the tests. Please add that.

- You are using `Board.new` in all of the test in **spec/board_spec.rb**. This is violating the DRY rule.

- Some of your test statements are ambiguous. That means they don't reflect the actual they are performing. The positive and negative scenario statements are the same. 

- If you have a couple of tests for a method, group them together with a `describe` block.

_I have added some comments below regarding the above review. Be sure to check them out._

Your project looks absolutely fantastic. Just make these changes and apply for another code review.

Happy coding 👨‍💻 

*Need help? Shoot me a message on Slack - [@Ghosh](https://app.slack.com/client/T47CT8XPG/DRDM6LAQN/user_profile/URB1MHXK6)*
<hr>

**Tirthajyoti Ghosh**

[GitHub](https://github.com/tirthajyoti-ghosh) | [LinkedIn](https://www.linkedin.com/in/tirthajyoti-ghosh/) | [Twitter](https://twitter.com/terrific_ghosh)

**Review by ElijahTobs — CHANGES_REQUESTED** [review]
# Status: Changes Required :recycle: 

Hi @Naqhid :wave: 

You've done well so far :+1:  and I commend you for that :clap: 
However, there are still some changes to be made before this milestone can be approved.

## REQUIRED

* [ ] Please, make all the changes as requested by the previous TSE.

## Happy Coding :computer: 

_Feel free to reach out to me on slack_ **`@Elijah Ayandokun`**
  **Elijah O. Ayandokun**
[Github](https://github.com/ElijahTobs) | [LinkedIn](https://www.linkedin.com/in/ayandokunelijah/) | [Twitter](https://twitter.com/ElijahTobs)

**Review by ElijahTobs — CHANGES_REQUESTED** [review]
# Status: Changes Required :recycle: 

Hi @Naqhid :wave: 

You've done well so far :+1:  and I commend you for that :clap: 
However, there are still some changes to be made before this milestone can be approved.

## REQUIRED
- [ ] Specs for each method should be wrapped in one `describe` block. As in example here https://github.com/rubocop-hq/rspec-style-guide#empty-lines-around-examples . ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/describe.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))


## Happy Coding :computer: 

_Feel free to reach out to me on slack_ **`@Elijah Ayandokun`**
  **Elijah O. Ayandokun**
[Github](https://github.com/ElijahTobs) | [LinkedIn](https://www.linkedin.com/in/ayandokunelijah/) | [Twitter](https://twitter.com/ElijahTobs)

**Review by mahmoud717 — APPROVED** [review]
### STATUS: APPROVED ✔️ ✔️ 

**hi @Naqhid **

**Great job with making all of the requirements. the project is approved, you can go ahead and merge this pull request**
